### PR TITLE
Delete account and relationships. 

### PR DESCRIPTION
**Pull request by jonathantorres**
Solves #154 
- Migration adds cascade on delete constraints on user relationships. 
- Added a test that confirms the fix.

### CODE REVIEW

File: app/Http/Controllers/AccountController.php [hunk 1]
```diff
@@ -94,11 +94,13 @@ public function delete()
 
     public function destroy()
     {
-        $user = Auth::user()->id;
-        $user->delete();
+        $userId = Auth::user()->id;
+        $user = User::find($userId);
```
Comment (mattstauffer) [inline app/Http/Controllers/AccountController.php]: Wow, that old code looks very broken. But I'm pretty sure we can just have this be `$user = Auth::user()`?

File: tests/AccountTest.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        $user = Factory::create('user');
+        $this->actingAs($user)
+             ->visit('account/delete')
+             ->press('Yes')
+             ->seePageIs('/')
+             ->see('Successfully deleted account.');
+
+        $this->dontSeeInDatabase('users', [
+            'email' => $user->email,
+        ]);
```
Comment (mattstauffer) [inline tests/AccountTest.php]: We'll probably want to test the cascade as well. but this is a great start--thanks!

File: app/Http/Controllers/AccountController.php (hunk 1 above)
Comment (jonathantorres) [inline app/Http/Controllers/AccountController.php]: yeah :+1:

**Comment by mattstauffer** [comment]
Well, I tried adding this to AppServiceProvider@boot:

```php
if (DB::connection() instanceof \Illuminate\Database\SQLiteConnection) {
    DB::statement(DB::raw('PRAGMA foreign_keys=1'));
}
```

... in order to enable foreign keys in SQLite. However, it broken everything; I'm guessing we have tests that insert some data without correct foreign key references. Or something.

Regardless, I wonder whether the better solution isn't just to explicitly delete all of the child/related items by using model events and listening to the deleted event: http://laravel-tricks.com/tricks/cascading-deletes-with-model-events

**Comment by jonathantorres** [comment]
Would you consider using MySQL for the tests? I can help you set that up if you want (in a separate PR of course)

**Comment by mattstauffer** [comment]
@jonathantorres Honestly, I think i'd prefer explicitly deleting in a `deleting` event over switching the stack. I think I'm just ready to commit: That's what I'd like to do with this. :)

**Comment by jonathantorres** [comment]
Sounds good! :+1:

**Comment by mattstauffer** [comment]
@jonathantorres thank you so much for your work on this!

**Comment by jonathantorres** [comment]
Happy to help!

**Pull request merged**